sc/managers/configman/configman.go

```diff
@@ -0,0 +1,88 @@
+package configman
+
+import (
+	"context"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/spacecloud-io/space-cloud/managers/configman/connector"
+	"github.com/spacecloud-io/space-cloud/model"
+	"go.uber.org/zap"
+)
+
+var connectorPool = caddy.NewUsagePool()
+var poolKey = "configman-poolkey"
+
+// ConfigMan manages all the database modules
+type ConfigMan struct {
+	// The config this app needs
+	StoreType string `json:"type,omitempty"`
+	Path      string `json:"path,omitempty"`
+
+	// For internal usage
+	logger     *zap.Logger
+	Connectors connector.ConfigManConnector
+}
+
+// CaddyModule returns the Caddy module information.
+func (ConfigMan) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "configman",
+		New: func() caddy.Module { return new(ConfigMan) },
+	}
+}
+
+// Provision sets up the file loader module.
+func (l *ConfigMan) Provision(ctx caddy.Context) error {
+	l.logger = ctx.Logger(l)
+
+	return nil
+}
+
+// Start begins the configman app operations
+func (l *ConfigMan) Start() error {
+	val, _, err := connectorPool.LoadOrNew(poolKey, func() (caddy.Destructor, error) {
+		return connector.New(l.logger, l.StoreType, l.Path)
+	})
+	if err != nil {
+		l.logger.Error("Unable to open configman connector", zap.String("store-type", l.StoreType), zap.String("path", l.Path), zap.Error(err))
+		return nil
+	}
+	l.Connectors = val.(*connector.Connector).Connector
```

Comment:
You shoud create a function in `connector.ConfigManConnector` to set the logger. Set the logger here because the logger may change on every start